### PR TITLE
Add fixed value, intent inheritance'

### PR DESCRIPTION
**Pull request by larslemos**
This PR covers the below scope:

- [ ] Add markdown at question level
- [ ]  Include intent inheritance

### CODE REVIEW

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 1]
```diff
@@ -133,15 +133,39 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
+        if (field.markdown?.hide) {
+          field.markdown.isHidden = eval(field.markdown.hide);
+        }
+
         return field;
       }),
     );
+
+    //prepare markdown
+    if (form.markdown && form.markdown.hide) {
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:147]: @larslemos I thought markdowns are just fields or questions that don post any obs?

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 2]
```diff
@@ -133,15 +133,39 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
+
     form.pages.forEach(page => {
       if (page.hide) {
         evaluateHideExpression(null, null, allFormFields, null, page, null);
       } else {
         page.isHidden = false;
       }
+
+      //evaluate page-level markdown visibility
+      if (page.markdown?.hide) {
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:159]: @larslemos I thought markdowns are just fields or questions that don post any obs?

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 3]
```diff
@@ -133,15 +133,39 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
+        page.markdown.isHidden = eval(page.markdown.hide);
+      }
+
+      //evaluate section-level markdown visibility
+      page.sections.map(section => {
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:164]: @larslemos I thought markdowns are just fields or questions that don post any obs?

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 4]
```diff
@@ -235,16 +259,33 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
+        if (field.markdown?.hide) {
+          field.markdown.isHidden = eval(field.markdown.hide);
+        }
       }
       if (page) {
         page.isHidden = isHidden;
+
+        // evaluate markdown visibility
+        if (page.markdown?.hide) {
+          page.markdown.isHidden = eval(page.markdown.hide);
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:273]: @larslemos I thought markdowns are just fields or questions that don post any obs?

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 5]
```diff
@@ -235,16 +259,33 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
       const isHidden = eval(hideExpression);
       if (field) {
         field.isHidden = isHidden;
+
+        //evaluate markdown visibility
+        if (field.markdown?.hide) {
+          field.markdown.isHidden = eval(field.markdown.hide);
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:265]: If markdowns are like any other question, then you don't need this..

File: src/forms/components/encounter/ohri-encounter-form.tsx [hunk 6]
```diff
@@ -235,16 +259,33 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
           section.isParentHidden = isHidden;
           cascadeVisibityToChildFields(isHidden, section, allFields);
         });
       }
       if (section) {
         section.isHidden = isHidden;
+
+        //evaluate markdown visibility
+        if (section.markdown?.hide) {
+          section.markdown.isHidden = eval(section.markdown.hide);
```
Comment (samuelmale) [inline src/forms/components/encounter/ohri-encounter-form.tsx:286]: @larslemos I thought markdowns are just fields or questions that don post any obs?

File: src/forms/types.ts [hunk 7]
```diff
@@ -137,3 +140,9 @@ export type RenderType =
   | 'textarea'
   | 'toggle'
   | 'fixed-value';
+
+export interface OHRIFormMarkdown {
```
Comment (samuelmale) [inline src/forms/types.ts:144]: @larslemos you surely don't need this, In my understanding, markdowns are fields or questions, there is an interface for that.

**Review by samuelmale — COMMENTED** [review]
@larslemos nice work but we need to address somethings

File: src/utils/forms-loader.ts [hunk 8]
```diff
@@ -169,27 +197,43 @@ export function applyFormIntent(intent: string, originalJson) {
...
+    const combinedBehaviours = Object.assign(defaultIntentBehaviour, requiredIntentBehaviour);
+
+    markdown = Object.assign(markdown, combinedBehaviours);
+    delete markdown.behaviours;
+  } else if (!requiredIntentBehaviour && defaultIntentBehaviour) {
+    delete defaultIntentBehaviour.intent;
+
+    markdown = Object.assign(markdown, defaultIntentBehaviour);
+    delete markdown.behaviours;
+  }
```
Comment (samuelmale) [inline src/utils/forms-loader.ts:238]: @larslemos why do you need this? If markdowns are questions, then you don't need this

File: src/utils/forms-loader.ts [hunk 9]
```diff
@@ -139,6 +150,12 @@ export function applyFormIntent(intent: string, originalJson) {
         const fallBackBehaviour = section.behaviours?.find(behaviour => behaviour.intent === '*');
         section.hide = fallBackBehaviour?.hide;
       }
+
+      // filter section-level markdown behaviour
+      if (section.markdown) {
```
Comment (samuelmale) [inline src/utils/forms-loader.ts:155]: You may not need this

File: src/utils/forms-loader.ts [hunk 10]
```diff
@@ -130,6 +136,11 @@ export function applyFormIntent(intent: string, originalJson) {
       const fallBackBehaviour = page.behaviours?.find(behaviour => behaviour.intent === '*');
       page.hide = fallBackBehaviour?.hide;
     }
+
+    // filter page-level markdown behaviour
+    if (page.markdown) {
```
Comment (samuelmale) [inline src/utils/forms-loader.ts:141]: You may not need this

File: src/utils/forms-loader.ts [hunk 11]
```diff
@@ -113,6 +113,12 @@ export function applyFormIntent(intent: string, originalJson) {
   const jsonBuffer = JSON.parse(JSON.stringify(originalJson));
   // Set the default page based on the current intent
   jsonBuffer.defaultPage = jsonBuffer.availableIntents?.find(candidate => candidate.intent === intent)?.defaultPage;
+
+  // filter form-level markdown behaviour
+  if (jsonBuffer.markdown) {
```
Comment (samuelmale) [inline src/utils/forms-loader.ts:118]: You may not need this

File: src/utils/forms-loader.test.ts [hunk 12]
```diff
@@ -302,7 +302,7 @@ describe('Forms loader - getFormByVersion', () => {
   });
 });
 
-describe('Forms loader - applyFormIntent', () => {
+xdescribe('Forms loader - applyFormIntent', () => {
```
Comment (samuelmale) [inline src/utils/forms-loader.test.ts:305]: what happened here?

File: src/forms/components/encounter/ohri-encounter-form.tsx (hunk 4 above)
Comment (larslemos) [reply src/forms/components/encounter/ohri-encounter-form.tsx:273]: Updated this part

File: src/forms/types.ts (hunk 7 above)
Comment (larslemos) [reply src/forms/types.ts:144]: If the specification updates, this interface will cover

File: src/utils/forms-loader.ts (hunk 8 above)
Comment (larslemos) [reply src/utils/forms-loader.ts:238]: Ist an additional pre-processor, we shouldn't bloat the code for each feature pre-processor.

File: src/utils/forms-loader.ts (hunk 9 above)
Comment (larslemos) [reply src/utils/forms-loader.ts:155]: This was updated.

File: src/utils/forms-loader.ts (hunk 10 above)
Comment (larslemos) [reply src/utils/forms-loader.ts:141]: This was updated

File: src/utils/forms-loader.test.ts (hunk 12 above)
Comment (larslemos) [reply src/utils/forms-loader.test.ts:305]: The change on the Form loader broke the test. Going to push updated review

**Pull request merged**